File: CheckBST.java
boolean isBST(Node root) {
        // code here.
        return bstUtil(root, Integer.MIN_VALUE, Integer.MAX_VALUE);
    }
    
    boolean bstUtil(Node root, int min, int max){
        if(root == null){
            return true;
        }
        
        if(root.data < min || root.data > max){
            return false;
        }
        
        
        boolean l = bstUtil(root.left, min, root.data);
        boolean r = bstUtil(root.right, root.data, max);
        return l && r;
    }

File: CheckBST.cpp
    bool bstUtils(Node* root, int min, int max) {
        if (root == nullptr) {
            return true;
        }
        
        if (root->data < min || root->data > max) {
            return false;
        }
        
        bool lSts = bstUtils(root->left, min, root->data);
        bool rSts = bstUtils(root->right, root->data, max);
        
        return lSts && rSts;
    }
    
    bool isBST(Node* root) {
        return bstUtils(root, INT_MIN, INT_MAX);
    }
